Vk typescript example (#378)
saucelabs/sauce-docs: docs/testrunner-toolkit/typescript.md

sidebar_label: Using TypeScript
---

> If you wrote your automated tests using TypeScript in either Playwright ,or Puppeteer, you need to transpile your Typescript files to Javascript before running them with Testrunner Toolkit.

Reviewer: ```suggestion
If you wrote your automated tests using TypeScript in either Playwright or Puppeteer, you need to transpile your Typescript files to Javascript before running them with `saucectl`.
```

> If you wrote your automated tests using TypeScript in either Playwright ,or Puppeteer, you need to transpile your Typescript files to Javascript before running them with Testrunner Toolkit.

Reviewer: ```suggestion
If you wrote your automated tests using TypeScript in either Playwright or Puppeteer, you need to transpile your Typescript files to Javascript before running them with Testrunner Toolkit.
```

   >
   > For more information on how to properly configure `tsconfig.json` please visit the [documentation](https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html#writing-a-configuration-file).

4. Next, edit the `suites` fields in `.sauce/config.yml` in order to ignore the `.ts` files and instead place the `.js` files inside the Testrunner Toolkit container:
    
   ```yaml
   suites:
     - name: "basic test"
       testMatch: 'tests/*.js'
   ```
   
   By default `saucectl` will pickup any `.js` files located in the designated directory, however with the `suites` field you can set more granular control with regular expressions.

> 5. Finally, run `saucectl` to execute your TypeScript tests:

Reviewer: This statement seems a bit confusing to me....aren't you technically running the JavaScript tests you just transpiled, NOT the Typescript tests?

Author: You are right, we are running JavaScript tests. I left this part as it was before. How do to suggest rewriting it?

> 5. Finally, run `saucectl` to execute your TypeScript tests:

Reviewer: ```suggestion
5. Finally, run `saucectl` to execute your transpiled tests:
```

Good?

Author: Yeap

Author: done
